File: java/Stack/reverse.java
import java.util.Scanner;
import java.util.Stack;
import java.util.Collections;

public class StackReverse {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Stack<Integer> stack = new Stack<>();

        System.out.print("Enter number of elements to push: ");
        int n = sc.nextInt();
        System.out.println("Enter elements:");
        for (int i = 0; i < n; i++) stack.push(sc.nextInt());

        System.out.println("Original Stack: " + stack);

        Collections.reverse(stack); // brutal shortcut
        System.out.println("Reversed Stack: " + stack);
    }
}

File: C/Stack/reverse.c
#include <stdio.h>
#define MAX 100

int stack[MAX];
int top=-1;

void push(int x) { if(top==MAX-1) { printf("Overflow!\n"); return; } stack[++top]=x; }

void display() {
    if(top==-1) { printf("Stack is empty\n"); return; }
    printf("Stack elements: ");
    for(int i=top;i>=0;i--) printf("%d ",stack[i]);
    printf("\n");
}

void reverse() {
    int start=0, end=top;
    while(start<end) {
        int temp=stack[start]; stack[start]=stack[end]; stack[end]=temp;
        start++; end--;
    }
}

int main() {
    int n, x;
    printf("Enter number of elements: "); scanf("%d",&n);
    for(int i=0;i<n;i++) { scanf("%d",&x); push(x); }
    printf("Original Stack:\n"); display();
    reverse();
    printf("Reversed Stack:\n"); display();
    return 0;
}
